### ISSUE TITLE
Feature Request: Fuzzy Search or Regex Support for Twingate Resources in Data Block

### ISSUE THREAD

**Issue by TylerStaplerAtFanatics** [Twingate/terraform-provider-twingate#356]
## Description:
Currently, when utilizing the twingate_resources data block in Terraform, there's no built-in support for fuzzy search or regex matching for Twingate resources in the account. We'd like to request a feature that allows for more flexible matching of resource names, which would be extremely helpful in use cases where resources follow a specific naming convention or pattern, and we need to fetch resources based on these patterns.

## Use case:
Consider a scenario where resources are named based on a certain structure or convention (e.g., including specific strings or codes in the names). We want to look up the resources in a separate terraform stack so that we can create a restricted service account.

Currently, we're manually listing out all possible strings/codes, and then using these to fetch the resources. However, this approach is not scalable or efficient as the number of resources grows or the naming conventions change.

Here's a simplified, genericized example of how we're currently working around this limitation:

```terraform
locals {
  all_codes = ["c", "a", "b", "d", "e"]
}

data "twingate_resources" "domain" {
  for_each = { for code in local.all_codes : code => code }
  name     = "fbg-${var.stage_name}-1${each.value} - unified"
}

resource "twingate_resource" "domain_access" {
  for_each = {
    for code, resource in data.twingate_resources.domain : code => resource
    if try(length(data.twingate_resources.domain[code].resources), 0) > 0
  }
  address           = each.value.resources[0].address
  name              = "fbg-${var.stage_name}-1${each.key} - Domain Access"
  remote_network_id = data.twingate_remote_network.network.id
  access {
    service_account_ids = [twingate_service_account.account[0].id]
  }
}
```

## Potential solution:
Add support for fuzzy search or regex matching in the twingate_resources data block, allowing users to fetch resources based on patterns or partial matches. This feature should be flexible enough to handle various matching scenarios and should be applicable to the name parameter.

Thank you for considering this feature request. Please let us know if there's any additional information we can provide.

**Comment by vmanilo** [comment]
@alexb-twingate we could do it in a few ways, one option could be to additional attribute`name_regexp` to allow search with regexp. Here is example of VMware datasource https://registry.terraform.io/providers/vmware/vcd/latest/docs/guides/data_source_filters

in our case it can be similar
```
data "twingate_resources" "domain" {
  name_regexp     = "fbg-*"
}
```

We can make them exclusive, so either `name` or `name_regexp` can be used. Same way we can add `name_starts_with`, `name_contains`, `name_not_equal`, `name_in`.

Alternatively, we can try to check if input value has predefined prefix like `regexp:`, `starts_with:`, `contains:`

```
data "twingate_resources" "domain" {
  name   = "regexp:fbg-*"
}
```

**Comment by alexb-twingate** [comment]
Fuzzy search for Resources will be resolved in #434 and we're adding filtering fields for everything else that can be queried using a string as well.

**Comment by bertekintw** [comment]
Closing this issue as it should be resolved with filtering enhancements introduced with 2.1.0 release.
